lib/doorkeeper/oauth/helpers/scope_checker.rb

```diff
@@ -6,7 +6,7 @@ def self.valid?(scope_str, server_scopes, application_scopes = nil)
           valid_scopes = if application_scopes.present?
                            server_scopes & application_scopes
                          else
-                           server_scopes
+                           OAuth::Scopes.new # Empty set
```

Comment:
We want existing applications with empty scopes to take into account only `server_scopes`, but this wouldn't allow that.